File: PAT/Basic Level/1015.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using std::cin;
using std::cout;
using std::endl;
using std::vector;

typedef struct {
    int ID;
    int dscore;
    int cscore;
    unsigned long long index;
}student;

bool order(const student & s1, const student & s2)
{
    if(s1.index > s2.index)
        return true;
    if(s1.index == s2.index)
    {
        if(s1.ID < s2.ID)
            return true;
    }
    return false;
}

int main()
{
    int n, low, high;
    cin >> n >> low >> high;
    vector<student> ddcc,ddc,dc,oth;
    for(int i = 0; i < n; i++)
    {
        int num,d,c;
        cin >> num >> d >> c;
        student xman;
        xman.cscore = c;
        xman.dscore = d;
        xman.ID = num;
        //xman.index = ((c + d) * 1000 + d + 1) * 100000000 - num;        // long long value not ok
        xman.index = (c + d) * 1000 + d;
        if(d >= low && c >= low)
            if(d >= high)
                if(c >= high)
                    ddcc.push_back(xman);
                else
                    ddc.push_back(xman);
            else
                if(d >= c)
                    dc.push_back(xman);
                else
                    oth.push_back(xman);
    }

    sort(ddcc.begin(), ddcc.end(), order);
    sort(ddc.begin(), ddc.end(), order);
    sort(dc.begin(), dc.end(), order);
    sort(oth.begin(), oth.end(), order);

    cout << ddcc.size() + ddc.size() + dc.size() + oth.size();
    for(vector<student>::iterator iter = ddcc.begin(); iter != ddcc.end(); iter++)
        cout << '\n' << iter->ID << " " << iter->dscore << " " << iter->cscore;
    for(vector<student>::iterator iter = ddc.begin(); iter != ddc.end(); iter++)
        cout << '\n' << iter->ID << " " << iter->dscore << " " << iter->cscore;
    for(vector<student>::iterator iter = dc.begin(); iter != dc.end(); iter++)
        cout << '\n' << iter->ID << " " << iter->dscore << " " << iter->cscore;
    for(vector<student>::iterator iter = oth.begin(); iter != oth.end(); iter++)
        cout << '\n' << iter->ID << " " << iter->dscore << " " << iter->cscore;

    return 0;
}

File: PAT/Basic Level/1015.py
class Student:
    def __init__(self, index, ID, dscore, cscore):
        self.index = index
        self.ID = ID
        self.dscore = dscore
        self.cscore = cscore

def compare(a, b):
    if a.index < b.index:
        return 1
    #if a.index == b.index:
    #    if a.ID > b.ID:
    #        return 1
    return -1

temp = raw_input().split()

number = int(temp[0])
lowline = int(temp[1])
highline = int(temp[2])

firstClass = []     # c > H && d > H
secondClass = []    # c < H && d > H
thirdClass = []     # c < H && d < H && d > c
finalClass = []     # 

for i in range(0, number):
    student = raw_input().split()
    ID = int(student[0])
    dScore = int(student[1])
    cScore = int(student[2])
    index = long(((dScore + cScore) * 1000 + dScore + 1) * 100000000 - ID)
    if dScore >= lowline and cScore >= lowline:
        xman = Student(index, ID, dScore, cScore)
        if dScore >= highline:
            if cScore >= highline:
                firstClass.append(xman)
            else:
                secondClass.append(xman)
        else:
            if dScore >= cScore:
                thirdClass.append(xman)
            else:
                finalClass.append(xman)

firstClass.sort(compare)
secondClass.sort(compare)
thirdClass.sort(compare)
finalClass.sort(compare)

print len(firstClass) + len(secondClass) + len(thirdClass) + len(finalClass)
for student in firstClass:
    print "%d %d %d" %(student.ID, student.dscore, student.cscore)
for student in secondClass:
    print "%d %d %d" %(student.ID, student.dscore, student.cscore)
for student in thirdClass:
    print "%d %d %d" %(student.ID, student.dscore, student.cscore)
for student in finalClass:
    print "%d %d %d" %(student.ID, student.dscore, student.cscore)           
